operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,980 @@
+            GridOrDensity::Density(params) => InitializedDensityRasterization::new(
+                vector_source,
+                out_desc,
+                tiling_specification,
+                params.radius,
+                params.stddev,
+            )
+            .map(InitializedRasterOperator::boxed),
+        }
+    }
+
+    span_fn!(Rasterization);
+}
+
+pub struct InitializedGridRasterization {
+    source: Box<dyn InitializedVectorOperator>,
+    result_descriptor: RasterResultDescriptor,
+    spatial_resolution: SpatialResolution,
+    grid_size_mode: GridSizeMode,
+    tiling_specification: TilingSpecification,
+    origin_coordinate: Coordinate2D,
+}
+
+impl InitializedRasterOperator for InitializedGridRasterization {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> util::Result<TypedRasterQueryProcessor> {
+        Ok(TypedRasterQueryProcessor::F64(
+            GridRasterizationQueryProcessor {
+                input: self.source.query_processor()?,
+                spatial_resolution: self.spatial_resolution,
+                grid_size_mode: self.grid_size_mode,
+                tiling_specification: self.tiling_specification,
+                origin_coordinate: self.origin_coordinate,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+pub struct InitializedDensityRasterization {
+    source: Box<dyn InitializedVectorOperator>,
+    result_descriptor: RasterResultDescriptor,
+    tiling_specification: TilingSpecification,
+    radius: f64,
+    stddev: f64,
+}
+
+impl InitializedDensityRasterization {
+    fn new(
+        source: Box<dyn InitializedVectorOperator>,
+        result_descriptor: RasterResultDescriptor,
+        tiling_specification: TilingSpecification,
+        radius: f64,
+        stddev: f64,
+    ) -> Result<Self, error::Error> {
+        ensure!(
+            radius > 0.,
+            error::InvalidOperatorSpec {
+                reason: "The radius for density rasterization must be strictly positive."
+                    .to_string()
+            }
+        );
+
+        Ok(InitializedDensityRasterization {
+            source,
+            result_descriptor,
+            tiling_specification,
+            radius,
+            stddev,
+        })
+    }
+}
+
+impl InitializedRasterOperator for InitializedDensityRasterization {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> util::Result<TypedRasterQueryProcessor> {
+        Ok(TypedRasterQueryProcessor::F64(
+            DensityRasterizationQueryProcessor {
+                input: self.source.query_processor()?,
+                tiling_specification: self.tiling_specification,
+                radius: self.radius,
+                stddev: self.stddev,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+pub struct GridRasterizationQueryProcessor {
```

Comment:
Maybe you can write a short doc-comment what this does.

Reply:
What exactly is there to say about it? It's more or less copied from other operators, where it is not commented as well.

Reply:
You probably meant the GridRasterization `raster_query` method and not the implementation of `InitializedRasterOperator`. It looked like your comment was referring to the latter because the conversation view gave the whole code block as context.
I described the method in general here. Lmk if it needs more detail. https://github.com/geo-engine/geoengine/pull/683/commits/cb3367da16b7a5ee740fbdd6c257b12767b1e446